packages/@uppy/status-bar/src/index.js

```diff
@@ -146,57 +146,40 @@ module.exports = class StatusBar extends Plugin {
 
     // TODO: move this to Core, to share between Status Bar and Dashboard
     // (and any other plugin that might need it, too)
-    const newFiles = Object.keys(files).filter((file) => {
-      return !files[file].progress.uploadStarted &&
-        !files[file].progress.preprocess &&
-        !files[file].progress.postprocess
-    })
-
-    const uploadStartedFiles = Object.keys(files).filter((file) => {
-      return files[file].progress.uploadStarted
-    })
-
-    const pausedFiles = uploadStartedFiles.filter((file) => {
-      return files[file].isPaused
-    })
 
-    const completeFiles = Object.keys(files).filter((file) => {
-      return files[file].progress.uploadComplete
-    })
+    const filesArray = Object.keys(files).map(file => files[file])
 
-    const erroredFiles = Object.keys(files).filter((file) => {
-      return files[file].error
+    const newFiles = filesArray.filter((file) => {
+      return !file.progress.uploadStarted &&
+        !file.progress.preprocess &&
+        !file.progress.postprocess
     })
 
-    const inProgressFiles = Object.keys(files).filter((file) => {
-      return !files[file].progress.uploadComplete &&
-             files[file].progress.uploadStarted
-    })
+    const uploadStartedFiles = filesArray.filter(file => file.progress.uploadStarted)
+    const pausedFiles = uploadStartedFiles.filter(file => file.isPaused)
+    const completeFiles = filesArray.filter(file => file.progress.uploadComplete)
+    const erroredFiles = filesArray.filter(file => file.error)
 
-    const inProgressNotPausedFiles = inProgressFiles.filter((file) => {
-      return !files[file].isPaused
+    const inProgressFiles = filesArray.filter((file) => {
+      return !file.progress.uploadComplete &&
+             file.progress.uploadStarted
     })
 
-    const startedFiles = Object.keys(files).filter((file) => {
-      return files[file].progress.uploadStarted ||
-        files[file].progress.preprocess ||
-        files[file].progress.postprocess
-    })
+    const inProgressNotPausedFiles = inProgressFiles.filter(file => !file.isPaused)
 
-    const processingFiles = Object.keys(files).filter((file) => {
-      return files[file].progress.preprocess || files[file].progress.postprocess
+    const startedFiles = filesArray.filter((file) => {
+      return file.progress.uploadStarted ||
+        file.progress.preprocess ||
+        file.progress.postprocess
     })
```

Comment:
I guess I can just use `startedFiles` in place of `uploadStarted`, since the first supersedes the last?

Reply:
I think you were right about this—testing a lot of files right now with rate limiting and it has some not-great results:
![image](https://user-images.githubusercontent.com/1006268/74664622-5ae71e00-519e-11ea-9753-f5d152a9c238.png)

Will address this in https://github.com/transloadit/uppy/pull/2060 most likely.